src/OaiPmh/HarvesterCommand.php

```diff
@@ -345,8 +346,14 @@ protected function execute(InputInterface $input, OutputInterface $output)
             try {
                 $this->harvestSingleRepository($input, $output, $target, $settings);
             } catch (\Exception $e) {
-                $this->writeLine($e->getMessage());
-                return 1;
+                if ($e instanceof OaiException
```

Comment:
I agree that it's strange to treat noRecordsMatch as an error, but since the OAI-PMH protocol treats it as such, I think it's okay for us to do the same for the sake of consistency.

Regarding a separate catch block for the OaiException, I did consider that, but since our special case relies not just on the exception class, but also on a specific code, I don't think I can do that without having to duplicate code, so this approach seemed a little more concise. If you can think of a more elegant approach, though, I'm open to suggestions!

Reply:
I think that's ok. I would have happily duplicated the lines, but I understand the desire to avoid that.